### PR TITLE
fix: tmp workaround for lack of ipfs peer id on status response

### PR DESCRIPTION
**Pull request by vasco-santos**
While we don't fix https://github.com/web3-storage/web3.storage/issues/1089 this was added to production

### CODE REVIEW

File: packages/api/src/utils/pin.js [hunk 1]
```diff
@@ -98,12 +98,19 @@ export function toPins (peerMap) {
   // Note: `clusterPeerId` is an internal id, and is only used for cluster admin.
   // The `ipfsPeerId` which we rename to `peerId` can be  used to connect to the underlying ipfs node
   // that stores a given pin, by passing it to `ipfs swarm connect <peerid>`.
-  return Object.entries(peerMap).map(([clusterPeerId, { peerName, status, ipfsPeerId: peerId }]) => {
+  const toPins Object.entries(peerMap).map(([clusterPeerId, { peerName, status, ipfsPeerId: peerId }]) => {
```
Comment (olizilla) [inline packages/api/src/utils/pin.js:101]:
```suggestion
  const toPins =  Object.entries(peerMap).map(([clusterPeerId, { peerName, status, ipfsPeerId: peerId }]) => {
```

File: packages/api/src/utils/pin.js [hunk 2]
```diff
@@ -98,12 +98,19 @@ export function toPins (peerMap) {
...
+  const toPins = Object.entries(peerMap).map(([clusterPeerId, { peerName, status, ipfsPeerId: peerId }]) => {
     return {
       status: toPinStatusEnum(status),
       location: { peerId, peerName }
     }
   })
+
+  const filteredToPins = toPins.filter(pin => pin.location?.peerId)
+  if (toPins.length !== filteredToPins.length) {
+    console.log('Dropping pins that do not have "peerId" on it', toPins.filter(pin => !!pin.location?.peerId))
```
Comment (alanshaw) [inline packages/api/src/utils/pin.js:110]: We need the `ipfsPeerId` we have no way to re-add the pins records after the upload.

**Review by alanshaw — CHANGES_REQUESTED** [review]

File: packages/api/src/utils/pin.js (hunk 2 above)
Comment (olizilla) [reply packages/api/src/utils/pin.js:110]: yes, but we currently fail an upload that is pinned on some cluster nodes if any one of them comes back with a missing `ipfsPeerId` property. Hector is taking a look on the cluster side, but we need to do something here too.

**Comment by vasco-santos** [comment]
Synced with @alanshaw we need to get this in and fix consequences later on.

**Comment by alanshaw** [comment]
The plan is to revert to using the cluster peer ID and _also_ store the IPFS peer ID. This will take a while to get together so lets get this merged and released and we'll have to follow up to add pin objects for the uploads later.

**Pull request merged**